### DISCUSSION TITLE
ESP32 stuck in Safeboot after OTA Ugrade from 13.4.0 to 14.0.0

### DISCUSSION THREAD

**Discussion by AlexKusnezov** [arendst/Tasmota#21420] [Support request] [answered]
Hey guys,

one of my esp32 can't get out of safeboot after trying to upgrade via ota.
Which version I should try to flash to get out of safeboot?
What am I missing?

**Comment by s-hadinger** [comment]
What is the version of your safeboot? What is happening?

**Reply by AlexKusnezov** [reply]
version `Tasmota 13.4.0(safeboot)`

see my reply to @sfromis

**Comment by sfromis** [comment]
You should flash a working binary for the ESP32 variant type you have, and ensure that you have a working URL, which Tasmota can reach. And if you have issues, make sure to read the console log for details.

Without flashing, you can also switch out of safeboot using the command `Restart 3`, but only if you have a working binary in the OTA partition. And you may not have that if you tried to flash the wrong binary.

**Reply by AlexKusnezov** [reply]
thanks, in therory this would be the working binary:
http://ota.tasmota.com/tasmota32/release-13.4.0/tasmota32.bin
so I paste the link in here:
<img width="401" alt="Screenshot 2024-05-16 at 18 01 26" src="https://github.com/arendst/Tasmota/assets/33514406/67b4d6c3-3731-4b15-97b0-12f353f8196c">

select Start Upgrade which gets me to this for a while:
<img width="521" alt="Screenshot 2024-05-16 at 18 01 11" src="https://github.com/arendst/Tasmota/assets/33514406/9187ea3b-8b7b-457b-9f9d-ac7503e0ecda">

after that it gets me to safeboot again

I feel like it hits some timeout on upload, because the connection isnt' very good. I've had that with some esp8266, which I solve by uploading a bin.gz version. 
Do you know by change how I can create a bin.gz for tasmota32.bin?

**Reply by sfromis** [reply]
If you created a `.gz`, that would be cause of such an error. This compression is *only* for ESP8266.

But again, check if the URL is accessible and **read the log** for symptoms.

**Reply by AlexKusnezov** [reply]
no, I did not such thing. Maybe I spot something in the logs

**Comment by Jason2866** [comment]
Just use the firmware you want. For esp32 no gzip is used. The provided firmwares from OTA server can be used for upload "by file" too.

**Comment by AlexKusnezov** [comment, marked as answer]
@sfromis @s-hadinger @Jason2866 Ok, that's actually a fun one: 
I use this esp32 with a couple of relays to trigger power and reboot of a computer, which also powers the esp32 directly from the motherboard. 
The whole damn time I was trying to update tasmota while the copmuter was off! 
As soon as I powered it on the and rebooted the esp it went straight to the new version.
Turns out the motherboard didn't provide sufficient power while it was off
